vision_agent_tools/models/florence2_sam2.py

```diff
@@ -66,95 +54,34 @@ def __init__(self, model_config: Florence2SAM2Config | None = None):
         )
         self.image_predictor = SAM2ImagePredictor(self.video_predictor)
 
-    def _dummy_agnostic_non_max_suppression(self, predictions, nms_threshold):
-        """
-        Apply agnostic Non-Maximum Suppression (NMS) to filter overlapping predictions.
-
-        Parameters:
-        predictions (dict[int, ImageBboxAndMaskLabel]): Dictionary of predictions.
-        nms_threshold (float): The IoU threshold value used for NMS.
-
-        Returns:
-        dict[int, ImageBboxAndMaskLabel]: Filtered predictions after applying NMS.
-        """
-        filtered_predictions = {}
-        prediction_items = list(predictions.items())
-
-        while prediction_items:
-            best_prediction = prediction_items.pop(0)
-            filtered_predictions[best_prediction[0]] = best_prediction[1]
-
-            prediction_items = [
-                pred
-                for pred in prediction_items
-                if self._calculate_iou(best_prediction[1].mask, pred[1].mask)
-                < nms_threshold
-            ]
-
-        return filtered_predictions
-
-    def _calculate_iou(
-        self, mask1: SegmentationBitMask, mask2: SegmentationBitMask
-    ) -> float:
-        """
-        Calculate the Intersection over Union (IoU) between two masks.
-
-        Parameters:
-        mask1 (numpy.ndarray): First mask.
-        mask2 (numpy.ndarray): Second mask.
-
-        Returns:
-        float: IoU value.
-        """
-        # Ensure the masks are binary
-        mask1 = mask1.astype(bool)
-        mask2 = mask2.astype(bool)
-
-        # Calculate the intersection and union
-        intersection = np.sum(np.logical_and(mask1, mask2))
-        union = np.sum(np.logical_or(mask1, mask2))
-
-        # Calculate the IoU
-        iou = intersection / union if union != 0 else 0
-
-        return iou
-
-    def _mask_to_bbox(self, mask: np.ndarray):
-        rows, cols = np.where(mask)
-        if len(rows) > 0 and len(cols) > 0:
-            x_min, x_max = np.min(cols), np.max(cols)
-            y_min, y_max = np.min(rows), np.max(rows)
-            return [x_min, y_min, x_max, y_max]
-
     def _update_reference_predictions(
         self,
-        last_predictions: dict[int, ImageBboxAndMaskLabel],
-        new_predictions: dict[int, ImageBboxAndMaskLabel],
+        last_predictions: dict[int, BboxAndMaskLabel],
+        new_predictions: dict[int, BboxAndMaskLabel],
```

Comment:
should we apply nms to this method?

Reply:
@hrnn we previously did but, as we migrated the NMS code from Florence2Sam2 into Florence2, the predictions given by Florence are now filtered and no double object predictions will be added. SAM2 will get in charge of propagating the input values given by Florence. This means that we no longer need to apply any NMS in the code of Florence2Sam2, we just need to pass through the `nms_threshold` input value into the Florence model.